core/controllers/profile.py

```diff
@@ -374,10 +451,27 @@ def get(self):
         self.render_template('oppia-root.mainpage.html')
 
 
-class SignupHandler(base.BaseHandler):
+class SignupHandlerNormalizedPayloadDict(TypedDict):
+    """Dict representation of SignupHandler's
+    normalized_payload dictionary.
+    """
+
+    username: str
+    agreed_to_terms: bool
+    default_dashboard: str
+    can_receive_email_updates: Optional[bool]
```

Comment:
Why can this be optional, that seems weird for a boolean.

Reply:
As far as I understand from the handler's code, I think keeping `can_receive_email_updates` as Optional is ok because we are deciding the execution of a particular piece of codebase on this condition https://github.com/oppia/oppia/blob/d7c0fb67e30cafa6d1d33d55128535ec36caadb5/core/controllers/profile.py#L445

I mean `user_services.update_email_preferences` can accept both false and true for its second argument.

currently, I'm looking at a case where the user wants to update_email_prefences but doesn't want to receive email updates, in that case, `can_receive_email_updates` can be false. But in a case where the user does not want to update email preferences then can_receive_email_updates can be None.

Reply:
Hmm, did you try looking at what we actually send from the frontend?

Reply:
From frontend, we are passing bool, https://github.com/oppia/oppia/blob/dcdf7f3cc131f8243240dbae3c6db6717e387f23/core/templates/pages/signup-page/services/signup-page-backend-api.service.ts#L37

But if I remove the default value `None` from here then all backend tests fail, because while testing we are never providing `can_receive_email_updates`:
 https://github.com/oppia/oppia/blob/dcdf7f3cc131f8243240dbae3c6db6717e387f23/core/tests/test_utils.py#L2452

Another possibility is that we should make the default value False and change this condition https://github.com/oppia/oppia/blob/dcdf7f3cc131f8243240dbae3c6db6717e387f23/core/controllers/profile.py#L445
to `if can_receive_email_updates:` but in this case this assertion fails https://github.com/oppia/oppia/blob/dcdf7f3cc131f8243240dbae3c6db6717e387f23/core/controllers/profile_test.py#L491

Reply:
TODO https://github.com/oppia/oppia/issues/16574 is added

Reply:
Hi @sahiljoster32 @vojtechjelinek -- if the frontend is always sending bool, then what is being proposed here (keeping it optional) doesn't make sense. We should just update the tests to always use that parameter, if the tests are wrong.

In general, please don't let the tests drive the code. Make sure to do the correct thing from the code perspective. If the tests need fixing, then so be it.

Reply:
`default_value` None is removed. In test_utils.py Now I'm using `feconf.DEFAULT_EMAIL_UPDATES_PREFERENCE`